controller/node_controller.go

```diff
@@ -637,14 +657,6 @@ func (nc *NodeController) syncDiskStatus(node *longhorn.Node) error {
 				diskUUID = diskConfig.DiskUUID
 			}
 			if diskStatusMap[id].DiskUUID == "" {
-				if len(disks) > 1 {
```

Comment:
It seems this fix has an issue. On line 649, we will create the disk configuration which is a file on the disk. That should be after we've verified there is no filesystem conflict.

Now if we move the check to a later place, the disk config (you can find it on the disk as `longhorn-disk.cfg`) might be created by mistake on the filesystem the disk shouldn't be valid.

Reply:
@yasker ,
I have move this check before the UUID check. But one question about this `longhorn-disk.cfg`, there are some checking after this file created. And I also don't see any deletion if we error out. I assume, maybe we clean this directory then it got deleted?

Reply:
1. I think you're referring to schedulable storage calculation later? Those won't impact the disk itself.
1. For deletion, `generateDiskConfig` should be atomic. There are two functions executed after https://github.com/longhorn/longhorn-manager/blob/master/util/util.go#L936 . I think we should add cleanup for failure then.